llvm/test/Transforms/LoopVectorize/dereferenceable-info-from-assumption-constant-size.ll

```diff
@@ -3,7 +3,7 @@
 
 declare void @llvm.assume(i1)
 
-define void @deref_assumption_in_header_constant_trip_count(ptr noalias %a, ptr noalias %b, ptr noalias %c) nofree {
+define void @deref_assumption_in_header_constant_trip_count(ptr noalias %a, ptr noalias %b, ptr noalias %c) nofree nosync{
```

Comment:
FWIW, I consider this a bug in our nofree semantics, and a large part of the reason why this never made progress :)

Frontends often have knowledge that a certain pointer is not going to be freed in the function (including via synchronization), but they don't have knowledge about whether it may be involved in synchronization. The current semantics make it impossible to convey such facts.